docs/training/distillation.md

```diff
@@ -0,0 +1,121 @@
+3. **Trains Models**: Runs forward passes on both models, but executes the backward pass only on the student model.
+4. **Saves Checkpoints**: Saves only the student model checkpoint, allowing it to be used later in the same manner as before.
+
+## Limitations
+
+* Only GPT-based checkpoints are currently supported.
+* Student and teacher models must support the same parallelism strategy.
+* If Pipeline Parallelism is enabled, intermediate-state based KD losses are only supported on the final pipeline stage.
+
+## Configuration
+
+### Knowledge Distillation Config
+
+You can configure the KD process via the `ModelOptDistillConfig` class or a YAML file. The configuration includes:
+
+* `logit_layers`: The layer names of student and teacher model logit layers. These names correspond to the PyTorch submodule attributes of the Megatron Core model. (For GPT-based models, this is `"output_layer"`). Default: `["output_layer", "output_layer"]`
+* `intermediate_layer_pairs`: A list of pairs of intermediate layer names. These pairs will by default have a Cosine-Similarity loss between them, and if tensor-parallelism is enabled, these layers must have sequence parallel outputs (i.e. LayerNorms), as Cosine loss cannot have a split hidden dimension. Default: `[["decoder.final_layernorm", "decoder.final_layernorm"]]`
+* `skip_lm_loss`: Whether to skip the default language modeling (LM) loss. If `false`, it will be added to the distillation loss. (Note it consumes more memory). Default: `true`
+* `kd_loss_scale`: Relative scale factor for the distillation loss. The cumulative logits-and-intermediate loss gets scaled to `kd_loss_scale` times the magnitude of the LM loss. Not used if `skip_lm_loss` is `true`. Default: `1.0`
+* `logit_kl_temperature`: Temperature variable for KL Divergence loss calculation. Default: `1.0`
+
+Example YAML configuration:
+
+```yaml
+logit_layers: ["output_layer", "output_layer"]
+intermediate_layer_pairs:
+  - ["decoder.final_layernorm", "decoder.final_layernorm"]
+logit_kl_temperature: 2.0
+```
+
+## Usage
+
+### Basic Usage with Default Configuration
+
+The simplest way to run knowledge distillation is to use or adapt one of the provided recipe scripts. Here's an example for distilling Llama3.2-3B into Llama3.2-1B:
+
+```bash
+torchrun --nproc_per_node=1 examples/recipes/llama/distill_llama32_3b-1b.py
+```
+
+### Using a Custom YAML Config File
+
+You can provide a custom YAML configuration file to override default settings:
+
+```bash
+torchrun --nproc_per_node=1 examples/recipes/llama/distill_llama32_3b-1b.py \
+    --config-file my_custom_config.yaml
+```
+
+### Using CLI Overrides
+
+Megatron Bridge supports Hydra-style CLI overrides for flexible configuration:
+
+```bash
+torchrun --nproc_per_node=2 examples/recipes/llama/distill_llama32_3b-1b.py \
+    model.tensor_model_parallel_size=2 \
+    model.teacher.tensor_model_parallel_size=2
+```
+
+### Combining YAML and CLI Overrides
```

Comment:
when would a user need to combine yaml config with CLI overrides? would it be cleaner to enforce that the user must update the yaml to change any parameter?